### PR TITLE
Added _binary support & REGEXP_LIKE

### PR DESCRIPTION
**Pull request by Hydrocharged**
Removed `REGEXP_MATCHES` as it does not exist in MySQL. It is a PostgreSQL function.

5.7 documentation: https://dev.mysql.com/doc/search/?d=12&p=1&q=regexp_matches
8.0 documentation: https://dev.mysql.com/doc/search/?d=201&p=1&q=regexp_matches
PostgreSQL: https://www.postgresql.org/docs/9.6/functions-matching.html

It was added before we forked the project, so I removed it. Not sure why it was ever added to begin with, as no MySQL client will ever expect it nor know how to interact with the results.

### CODE REVIEW

File: enginetest/insert_queries.go [hunk 1]
```diff
@@ -75,6 +75,18 @@ var InsertQueries = []WriteQueryTest{
 		SelectQuery:         "SELECT i FROM mytable WHERE s = 'x';",
 		ExpectedSelect:      []sql.Row{{int64(999)}},
 	},
+	{
+		WriteQuery:          "INSERT INTO mytable VALUES (999, _binary 'x');",
```
Comment (zachmu) [inline enginetest/insert_queries.go:79]: Is it easy to write a test where the _binary value of the string differs from the golang literal? This doesn't appear to be testing much as is

File: enginetest/queries.go [hunk 2]
```diff
@@ -3371,32 +3371,18 @@ var QueryTests = []QueryTest{
...
+		Expected: []sql.Row{
+			{int8(1)},
+		},
 	},
 	{
-		Query:    `SELECT REGEXP_MATCHES("", "", NULL)`,
-		Expected: []sql.Row{{nil}},
+		Query: "SELECT i, s, REGEXP_LIKE(s, '[a-z]+d row') FROM mytable;",
+		Expected: []sql.Row{
+			{int64(1), "first row", int8(0)},
```
Comment (zachmu) [inline enginetest/queries.go:3382]: No need to explicitly type these ints

File: sql/expression/function/regexp_like.go [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		}
+	case 2:
+		r = &RegexpLike{
+			Text:    args[0],
+			Pattern: args[1],
+		}
+	default:
+		return nil, sql.ErrInvalidArgumentNumber.New("regexp_like", "2 or 3", len(args))
+	}
+	if canBeCached(r.Pattern) {
```
Comment (zachmu) [inline sql/expression/function/regexp_like.go:57]: Can't do this now, you have to wait until execution so the expressions are all resolved

**Review by zachmu — APPROVED** [review]
LGTM, just a minor issue where caching the regexp won't work as intended

File: enginetest/insert_queries.go (hunk 1 above)
Comment (Hydrocharged) [reply enginetest/insert_queries.go:79]: I don't think so, our charset and collation story isn't robust enough. We would first have to define a difference between a binary string and utf8mb4 string, but currently we just see them both as standard strings.

File: sql/expression/function/regexp_like.go (hunk 3 above)
Comment (Hydrocharged) [reply sql/expression/function/regexp_like.go:57]: I retrofitted this from the `regexp_matches` code, I should have vetted this a bit more. Glad you caught it.

**Pull request merged**